### PR TITLE
Qty 10001: Add support for MT time by mapping to America/Denver

### PR DESCRIPTION
**Pull request by smpairingstations**
[Link to Jira ticket](https://strongmind.atlassian.net/browse/QTY-10001)

## Purpose 
User's are experiencing a bug where times in MDT are being converted to MST.

## Approach 
Add MT as a custom timezone that maps to ActiveSupport, America/Denver.

## Testing
Specs

### CODE REVIEW

File: app/decorators/models/course_decorator.rb [hunk 1]
```diff
@@ -144,12 +144,20 @@ def coalesce_datetime(date:, time:)
...
+    parsed_in_zone = parse_time_in_zone(time)
     hour_offset = (parsed_in_zone.utc_offset / 3_600) * -1
     rollover_limit = 24 - hour_offset
     parsed_in_zone.hour >= rollover_limit ? 1 : 0
   end
 
+  def parse_time_in_zone(time)
+    school_timezone = SettingsService.get_settings(object: 'school', id: 1)['timezone']
+    custom_timezones = {
+      'MT' => ActiveSupport::TimeZone['America/Denver'],
```
Comment (AliArehart) [inline app/decorators/models/course_decorator.rb:156]: Is this problem something that is actively only affected the MT timezone, or would it be affecting our DLS oberseving timezones?

**Comment by AliArehart** [comment]
How did you get around your issues testing timezone with your local configuration?

File: app/decorators/models/course_decorator.rb (hunk 1 above)
Comment (smpairingstations) [reply app/decorators/models/course_decorator.rb:156]: We believe that could be the case, but we have decided to take the problems as they come. If we have issues with other timezones, we should add them to custom_timezones.

**Review by Lendolan — APPROVED** [review]

**Comment by smpairingstations** [comment]
> How did you get around your issues testing timezone with your local configuration?

We did not, unfortunately. We timeboxed it and decided to not let it halt progress.

**Review by AliArehart — APPROVED** [review]

**Pull request merged**